pkg/operator/ceph/cluster/osd/pod.go

```diff
@@ -90,37 +91,49 @@ func (c *Cluster) makeDeployment(nodeName string, devices []rookalpha.Device, se
 		logger.Infof("no data dir provided")
 	}
 
-	// Mount the path to the directory-based osd unless it is already available under the dataDirHostPath
-	if osd.IsDirectory && !strings.HasPrefix(osd.DataPath, k8sutil.DataDir) {
-		volumeName := k8sutil.PathToVolumeName(osd.DataPath)
-		dataDirSource := v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: osd.DataPath}}
-		volumes = append(volumes, v1.Volume{Name: volumeName, VolumeSource: dataDirSource})
-		volumeMounts = append(volumeMounts, v1.VolumeMount{Name: volumeName, MountPath: osd.DataPath})
-	}
-
-	// Mount the required device
-	if !osd.IsDirectory {
+	// Mount the path to the directory-based osd
+	var dataDir string
+	if osd.IsDirectory {
+		// osd.DataPath includes the osd subdirectory, so we want to mount the parent directory
+		parentDir := filepath.Dir(osd.DataPath)
+		dataDir = parentDir
+		// skip the mount if this is the default directory being mounted, which will look like "/var/lib/rook", even if the dataDirHostPath is a different path
+		if parentDir != k8sutil.DataDir {
+			volumeName := k8sutil.PathToVolumeName(parentDir)
+			dataDirSource := v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: parentDir}}
+			volumes = append(volumes, v1.Volume{Name: volumeName, VolumeSource: dataDirSource})
+			configVolumeMounts = append(configVolumeMounts, v1.VolumeMount{Name: volumeName, MountPath: parentDir})
+		}
+		volumeMounts = configVolumeMounts
+	} else { // Mount the required device
+		dataDir = k8sutil.DataDir
 		// create volume config for the data dir and /dev so the pod can access devices on the host
 		devVolume := v1.Volume{Name: "devices", VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: "/dev"}}}
 		volumes = append(volumes, devVolume)
 		devMount := v1.VolumeMount{Name: "devices", MountPath: "/dev"}
-		volumeMounts = append(volumeMounts, devMount)
+		volumeMounts = append(configVolumeMounts, devMount)
 	}
 
 	if len(volumes) == 0 {
 		return nil, fmt.Errorf("empty volumes")
 	}
 
+	osdID := strconv.Itoa(osd.ID)
+	tiniEnvVar := v1.EnvVar{Name: "TINI_SUBREAPER", Value: ""}
```

Comment:
no const for this `TINI_SUBREAPER` env var?

Reply:
I've become intentionally more relaxed over time that when a const is used exactly once, it seems like overkill to define a const. Have you seen in a go style guide that they should be defined? I don't see it [here](https://golang.org/doc/effective_go.html#constants).